packages/sanity/src/desk/components/pane/PaneLayout.tsx

```diff
@@ -50,6 +51,16 @@ export function PaneLayout(
     if (!collapsed && onExpand) onExpand()
   }, [collapsed, onCollapse, onExpand])
 
+  // Force popover position update when panes change
+  // This is needed to ensure that any open popovers is positioned correctly if a pane is toggled
+  // between collapsed and expanded. We normally use the boundary element, but that won't work
+  // inside the PTE
+  useEffect(() => {
+    if (state.panes.length) {
+      forcePopoverUpdate?.()
+    }
```

Comment:
Would it make sense / is there a way of checking if there are even popups open? I can see that this could be called multiple times without necessarily needing it if no PTE ends up having any popovers open.

Or perhaps the `forcePopoverUpdate` (which might be better served with a more descriptive name) already checks for this case?

Reply:
There isn't a way currently, but there might be a way we can constrain this to only run if the document panel is resized.

For what it is worth, the only thing the method does is update the state in the context, and won't do anything unless any popovers is listening to the update.

Reply:
We could do something like this to prevent the update function from running when there are no popovers in the DOM:

```ts
function hasPopovers() {
  const attr = '[data-ui="Popover"]'
  const popovers = document.querySelectorAll(attr)
  return popovers.length > 0
}
```

```ts
  useEffect(() => {
    if (state.panes.length && hasPopovers()) {
      forcePopoverUpdate?.()
    }
  }, [state.panes, forcePopoverUpdate])

```